src/coverlet.msbuild.tasks/CoverageResultTask.cs

```diff
@@ -230,7 +248,12 @@ public override bool Execute()
                         exceptionMessageBuilder.AppendLine($"The {thresholdStat.ToString().ToLower()} method coverage is below the specified {_threshold}");
                     }
 
-                    throw new Exception(exceptionMessageBuilder.ToString());
+                    if(thresholdAct == ThresholdAction.Fail)
```

Comment:
I prefer another else if (dThresholdAct == ThresholdAction.Warning) or a switch, today we've 2 condition and should not change...but I want to be sure that semantic won't change if we add other states in future.

Reply:
changed to using a switch statement